cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -386,7 +391,16 @@ static PyObject *register_timed_callback(PyObject *self, PyObject *args)
 
     // Extract the time
     PyObject *pTime = PyTuple_GetItem(args, 0);
-    time_ps = PyLong_AsLongLong(pTime);
+    if (PyLong_Check(pTime)) {
+        time_ps = PyLong_AsUnsignedLongLong(pTime);
+#if PY_MAJOR_VERSION < 3
+    } else if (PyInt_Check(pTime)) {
+        time_ps = PyInt_AsUnsignedLongLongMask(pTime);
```

Comment:
As above: please add error handling

Reply:
Why can't the same code path be used for both of these cases?